skr/webhook/internal/handler_test.go

```diff
@@ -0,0 +1,145 @@
+package internal_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+
+	"github.com/kyma-project/kyma-watcher/skr/webhook/internal"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var (
+	testEnv   *envtest.Environment //nolint:gochecknoglobals
+	k8sClient client.Client        //nolint:gochecknoglobals
+)
+
+const (
+	uid = "someUid"
+)
+
+func getHTTPRequest(operation admissionv1.Operation, crdName string) (*http.Request, *httptest.ResponseRecorder) {
+	admissionReview, err := createAdmissionRequest(operation, crdName)
+	Expect(err).ShouldNot(HaveOccurred())
+	bytesRequest, err := json.Marshal(admissionReview)
+	Expect(err).ShouldNot(HaveOccurred())
+	req := httptest.NewRequest(http.MethodGet, "/validate", bytes.NewBuffer(bytesRequest))
+	w := httptest.NewRecorder()
+	return req, w
+}
+
+func createAdmissionRequest(operation admissionv1.Operation, crdName string) (*admissionv1.AdmissionReview, error) {
+	admissionReview := &admissionv1.AdmissionReview{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       admissionv1.AdmissionReview{}.Kind,
+			APIVersion: admissionv1.AdmissionReview{}.APIVersion,
+		},
+		Request: &admissionv1.AdmissionRequest{
+			Name: crdName,
+			Kind: metav1.GroupVersionKind{
+				Kind:    "Manifest",
+				Version: "v1alpha1",
+				Group:   "component.kyma-project.io",
+			},
+			Operation: operation,
+			UID:       uid,
+		},
+	}
+
+	objectWatched := &internal.ObjectWatched{
+		Metadata: internal.Metadata{
+			Name:      "manifestObj",
+			Namespace: metav1.NamespaceDefault,
+		},
+		Spec: map[string]interface{}{
+			"specField": "value",
+		},
+		Kind: "Manifest",
+	}
+
+	rawBytes, err := json.Marshal(objectWatched)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	// change watched field
+	objectWatchedOld := objectWatched
+	objectWatchedOld.Spec["specField"] = "oldValue"
+	rawBytesOld, err := json.Marshal(objectWatchedOld)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	admissionReview.Request.Object.Raw = rawBytes
+	admissionReview.Request.OldObject.Raw = rawBytesOld
+
+	return admissionReview, nil
+}
+
+var _ = Describe("Kyma with multiple module CRs", Ordered, func() {
+	configMap := v1.ConfigMap{}
+	kyma := unstructured.Unstructured{}
+	BeforeAll(func() {
+		// config map
+		configMapContent, err := os.Open("./assets/configmap.yaml")
```

Comment:
maybe create a small file helper for this kind of stuff

Reply:
more tests will be added and then common utils will be refactored